src/lamp_py/performance_manager/l1_rt_trips.py

```diff
@@ -517,314 +653,315 @@ def update_stop_sequence(db_manager: DatabaseManager) -> None:
-                    .scalar_subquery()
-                )
-            ),
-            sa.or_(
-                StaticRoutePatterns.route_pattern_typicality == 1,
-                StaticRoutePatterns.route_pattern_typicality == 5,
-            ),
-        )
-        .order_by(
-            sa.func.coalesce(
-                StaticTrips.branch_route_id, StaticTrips.trunk_route_id
-            ),
-            StaticRoutePatterns.direction_id,
-            StaticRoutePatterns.static_version_key,
-            StaticRoutePatterns.route_pattern_typicality.desc(),
-        )
-        .subquery("canon_trips")
-    )
-    # using the representative_trip_id's from the canon_trips query, create
-    # stop_sequence values for each parent_station on each route in each direction.
-    # stop_sequence's are created using the row_number function so that they
-    # always start at 1 and increment according to the
-    # StaticStopTimes.stop_sequence order
-    static_canon = (
-        sa.select(
-            canon_trips.c.direction_id,
-            canon_trips.c.trunk_route_id,
-            canon_trips.c.route_id,
-            StaticStops.parent_station,
-            sa.over(
-                sa.func.row_number(),
-                partition_by=(
-                    canon_trips.c.static_version_key,
-                    canon_trips.c.direction_id,
-                    canon_trips.c.route_id,
-                ),
-                order_by=StaticStopTimes.stop_sequence,
-            ).label("stop_sequence"),
-            canon_trips.c.static_version_key,
-        )
-        .select_from(canon_trips)
-        .join(
-            StaticStopTimes,
-            sa.and_(
-                canon_trips.c.representative_trip_id == StaticStopTimes.trip_id,
-                canon_trips.c.static_version_key
-                == StaticStopTimes.static_version_key,
-            ),
-        )
+    distinct_query = (
+        sa.select(VehicleTrips.service_date, VehicleTrips.static_version_key)
+        .distinct()
         .join(
-            StaticStops,
-            sa.and_(
-                StaticStopTimes.stop_id == StaticStops.stop_id,
-                StaticStopTimes.static_version_key
-                == StaticStops.static_version_key,
-            ),
+            TempEventCompare,
+            VehicleTrips.pm_trip_id == TempEventCompare.pm_trip_id,
         )
-        .subquery("static_canon")
     )
-
-    # subquery to join static_canon results to vehicle_events records
-    rt_canon = (
-        sa.select(
-            VehicleEvents.pm_event_id,
-            static_canon.c.stop_sequence,
-        )
-        .select_from(VehicleEvents)
-        .join(
-            VehicleTrips,
-            VehicleEvents.pm_trip_id == VehicleTrips.pm_trip_id,
-        )
-        .join(
-            static_canon,
-            sa.and_(
-                VehicleTrips.direction_id == static_canon.c.direction_id,
+    for record in db_manager.select_as_list(distinct_query):
```

Comment:
These queries are the exact same, except running in a for loop with specific `service_date` and `static_version_key` values in the where clauses. In testing this dramatically improved the performance of these queries from ~100 seconds to < 5 seconds locally.